### PR TITLE
Dapp bridge

### PR DESCRIPTION
**Pull request by samsiegart**
refs: https://github.com/Agoric/agoric-sdk/issues/6294

Works with https://github.com/Agoric/agoric-sdk/pull/6486 and `yarn link @agoric/web-components`

### Screenshots:
Network picker:
image

### CODE REVIEW

**Review by turadg — COMMENTED** [review]
Nice simplification. I'll do a more thorough review after the SDK side firms up

**Comment by samsiegart** [comment]
This is blocked by https://github.com/Agoric/agoric-sdk/pull/6463, after that deploys I'll have to update `package.json` here to point to the latest sdk package versions

File: src/components/KeplrConnection.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  keplrConnectionAtom,
+  networkConfigAtom,
+} from 'store/app';
+import { watchContract, watchPurses } from 'utils/updates';
+import NetworkDropdown from 'components/NetworkDropdown';
+
+import 'styles/globals.css';
+import clsx from 'clsx';
+
+const KeplrConnection = () => {
```
Comment (turadg) [inline src/components/KeplrConnection.tsx:27]: this isn't really about Keplr. Keplr is for signing. This should work without Keplr if given an address and network config. consider `NetworkConnection` or `NetworkConfigurator`.

the only thing dapp-psm should use Keplr for is to get the address.

File: src/services/swap.tsx [hunk 2]
```diff
@@ -62,43 +62,46 @@ const makeSwapOffer = ({
...
-      ...defaultToastProperties,
-      type: toast.TYPE.INFO,
-      progress: undefined,
-      hideProgressBar: true,
-      autoClose: false,
-    })
+  setTimeout(() => {
+    setSwapped(false);
+  }, 3000);
+  toast.success(
```
Comment (turadg) [inline src/services/swap.tsx:92]: very odd for a _service_ to be JSX. Looks like this is the only reason to make this service .tsx. Instead please move this component to another file and have this call toast or component create from there

File: src/vite-env.d.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
 /// <reference types="vite/client" />
 
+declare module '@agoric/web-components' {
+  export const makeAgoricKeplrConnection;
```
Comment (turadg) [inline src/vite-env.d.ts:4]: consider copying this over to make refactoring easier. and so you don't rely on agoric-sdk releases.

File: src/components/KeplrConnection.tsx (hunk 1 above)
Comment (samsiegart) [reply src/components/KeplrConnection.tsx]: Renamed to `ChainConnection`

File: src/services/swap.tsx (hunk 2 above)
Comment (samsiegart) [reply src/services/swap.tsx]: Yea I think this was a clue that some concerns needed to be separated. Just changed this service to be more decoupled from UI state

File: src/vite-env.d.ts (hunk 3 above)
Comment (samsiegart) [reply src/vite-env.d.ts:4]: I still need the @agoric/web-components/react component for the bridge so I'll just leave this for now

File: package.json [hunk 4]
```diff
@@ -22,10 +22,8 @@
   "devDependencies": {
     "@agoric/casting": "dev",
     "@agoric/ertp": "^0.14.2",
-    "@agoric/notifier": "^0.4.0",
     "@agoric/ui-components": "^0.2.33",
-    "@agoric/wallet": "^0.13.1",
-    "@agoric/web-components": "0.3.1-dev-8ac3c4c.0",
+    "@agoric/web-components": "^0.5.0",
```
Comment (samsiegart) [inline package.json:26]: Will upgrade to 0.6.0 once https://github.com/Agoric/agoric-sdk/pull/6486 lands then the build should work

Comment (mhofman) [reply package.json:26]: A 0.6.0 wouldn't be published until a new release of the SDK is published.

I'd recommend you look into using prerelease versions that are automatically built and pushed to npm.

File: src/components/WalletBridge.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          {message && `: ${message}`}
+        </p>
+      </div>
+    );
+  };
+
+  const onBridgeMessage = (ev: any) => {
+    const data = ev.detail?.data;
+    const type = data?.type;
+    switch (type) {
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply src/components/WalletBridge.tsx]: I don't think it needs to be that strict as to throw, it doesn't necessarily mean the bridge is broken if we add another message that the dapp doesn't care about. Added some types though.

File: src/config.ts [hunk 6]
```diff
@@ -3,3 +3,30 @@ export const dappConfig = {
   INSTANCE_PREFIX: ':published.psm.IST.',
   INSTANCES_KEY: ':published.agoricNames.instance',
 };
+
+export const networkConfigs = {
```
Comment (turadg) [inline src/config.ts:7]: we'll want to import these eventually

**Review by turadg — APPROVED** [review]

**Pull request merged**